xarray/core/types.py

```diff
@@ -64,7 +64,8 @@
 VarCompatible = Union["Variable", "ScalarOrArray"]
 GroupByIncompatible = Union["Variable", "GroupBy"]
 
-Dims = Union[str, Iterable[Hashable], None]
+Dims = Union[str, Iterable[Hashable], "ellipsis", None]
```

Comment:
This changes behaviour in _aggregations.py:

https://github.com/pydata/xarray/blob/0dfcfddf3ea5a0f54115dcfb9cb1edfeb0a281c7/xarray/core/_aggregations.py#L2052-L2054

https://github.com/pydata/xarray/blob/0dfcfddf3ea5a0f54115dcfb9cb1edfeb0a281c7/xarray/core/_aggregations.py#L3442-L3444

Maybe it's ok?